torch/csrc/cuda/Event.cpp

```diff
@@ -45,21 +45,21 @@ static PyObject * THCPEvent_pynew(
 static PyObject * THCPEvent_from_ipc_handle(
     PyTypeObject *type, PyObject *args) {
   HANDLE_TH_ERRORS
-  long long device_index = -1;
+  THPDevice * device = nullptr;
   const char *handle_bytes = nullptr;
   int handle_size = 0;
 
   // cannot use bool 'p' and bytearray 'Y' as they are not available in Python 2
   if (!PyArg_ParseTuple(
-      args, "Ls#", &device_index, &handle_bytes, &handle_size)) {
+      args, "Os#", &device, &handle_bytes, &handle_size)) {
```

Comment:
This is missing a check that the first argument is actually a `THPDevice`. I think this is complicated enough that it might be worth using `PythonArgParser`, but it's up to you. `tensor_new.cpp` has a bunch of examples. Something like:

```c++
static PythonArgParser parser({
  "from_ipc_handle(Device device, std::string ipc_handle)",
});
ParsedArgs<2> parsed_args;
auto r = parser.parse(args, kwargs, parsed_args);
...
```

One additional advantage is that it handles the conversion from strings and ints to Device consistently with our other functions.

Reply:
Thanks, let me try.